xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -366,10 +354,9 @@ private static int getMatchingPrefixLength(
         InetAddress cidrAddr = cidrRange.getAddressPrefix();
         boolean cidrIsIpv6 = cidrAddr instanceof Inet6Address;
         if (isIPv6 == cidrIsIpv6) {
-          byte[] cidrBytes = cidrAddr.getAddress();
           int prefixLen = cidrRange.getPrefixLen();
-          if (isCidrMatching(cidrBytes, addressBytes, prefixLen)
-              && prefixLen > matchingPrefixLength) {
+          CidrMatcher matcher = CidrMatcher.create(cidrAddr, prefixLen);
```

Comment:
It is OK for now. But I think a better approach is to model `CidrMatcher`s in [`FilterChainMatch`](https://github.com/grpc/grpc-java/blob/fa4b980e07400a66cfaf3693785f873031b20482/xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java#L226-L228) directly at the time it was validated and parsed.

But since I am also doing a significant refactoring for these data classes, it may be painful to coordinate conflicts. So it is fine to do this trick here now.